### ISSUE TITLE
Incorrect text when copying from pdf and pasting into a text editor

### ISSUE THREAD

**Issue by EvgeniyBudaev** [diegomura/react-pdf#3009]
**Describe the bug**
We connect the font via Font.register. Text 123ff456fi987. Copy in the downloaded pdf file 123ff456fi987. Paste it into a text editor. We get 123f456f987 instead of 123ff456fi987. Bottom line: there is a bug with the symbols ff, fi with plug-in fonts.

**To Reproduce**
Steps to reproduce the behavior including code snippet (if applies):

1. https://github.com/EvgeniyBudaev/react-pdf-renderer
2. Dowload pdf file with text: 123ff456fi987
3. Copy in the downloaded pdf file 123ff456fi987
4. Paste it into a text editor.
5. 5. We get 123f456f987 instead of 123ff456fi987

_You can make use of [react-pdf REPL](https://react-pdf.org/repl) to share the snippet_

**Expected behavior**
123ff456fi987

**Screenshots**

**Desktop (please complete the following information):**

- OS: [Windows, Ubuntu]
- Browser [chrome]
- React-pdf version [e.g. v4.1.6]
![2024-12-27_14-50_1](https://github.com/user-attachments/assets/eda7a915-69bb-4df8-a3ae-7169ef1381f5)
![2024-12-27_14-50](https://github.com/user-attachments/assets/25d3596a-f9a7-4481-91c3-36bcf9cde8bf)

**Comment by VladSez** [comment]
I have the same problem with registered fonts (Open Sans) when copying from pdf

```
 "@react-pdf/layout": "4.4.0",
 "@react-pdf/renderer": "4.3.0",
```

**Comment by VladSez** [comment]
@diegomura created a repro and video with a bug

https://react-pdf.org/repl?code=3187b0760ce02e00408a057025803c450298c0bc300500943807cf805030c00f0022230080b6198531165d400a02180e659a004f003619b006f61622003a0046200099080beed3a72a005430a5830019aa0c4bd805100b48c7921100b88d2303cc300c000430dbbb34230e4d2a007a5d7d0d4d1840ea60de010890ba0666567602006e3232003170285900270c3e24680c02bc090e431e463b214700720079000716180065570846801a0e0802e026800b282856887b60e0e0253059002b08250c1124003702d9300c2860b056c6609076b04b081ecb1e11110f000659004659006660c33c8853f3cbeba29f010221e0ec26864699154192ca8120b06906020382e94144184ea8c30fb3f11478984a871142a4715534ad1e12894ce3e368ce8e378015806e4ca752c07c0010ba0a02046238006ccce8852a93480048800a4800179e56e0ce1668beac6cad5ea4d36875ba9048651540368664c800250c0f180502e0d1b285161ac2a54442a1d05860b106140000

https://github.com/user-attachments/assets/299791da-4662-498e-b514-a0c5e392d104

**Comment by mauricedoepke** [comment]
I think this is a bug in fontkit which is used by react-pdf.
See https://github.com/foliojs/fontkit/issues/154#issuecomment-4187174366 For my suspected reason and a workaround including a patch file.

**Comment by diegomura** [comment]
I'm not being able to reproduce it anymore as it shows [here](https://github.com/diegomura/react-pdf/issues/3009#issuecomment-2708484766). We had a PR fixing this awhile ago. I'll close it until further notice

**Comment by VladSez** [comment]
Yes, make sense. I also couldn’t reproduce the bug recently. So probably was fixed. Thank you
